salt/journal-cms/processes.sls

```diff
@@ -30,27 +30,16 @@ journal-cms-processes-start:
 {% set controller = "journal-cms-processes" %}
 
 {{ controller }}-script:
-    file.managed:
+    file.absent:
         - name: /opt/{{ controller }}.sh
-        - source: salt://elife/config/etc-init-multiple-processes-parallel.conf
-        - template: jinja
-        - context:
-            processes: {{ processes }}
-            timeout: 70
-
-{{ controller }}-service:
-    file.managed:
-        - name: /lib/systemd/system/{{ controller }}.service
-        - source: salt://journal-cms/config/lib-systemd-system-{{ controller }}.service
 
+{% for process, num_processes in processes.items() %}
+{{ process }}-service:
     service.running:
-        - name: {{ controller }}
+        - name: {{ process }}@{1..{{ num_processes }}}
+        - enable: true
         - require:
-            - file: {{ controller }}-service
-            - {{ controller }}-script
-            {% for process in processes %}
             - file: {{ process }}-init
-            {% endfor %}
-            
```

Comment:
the existing script always restarts the service, and I haven't found a way to make `service.running` do this as it would require lots of `watch` statements and still that would only restart them upon the watches detecting changes e.g. in the code. So this part is not complete.

Reply:
I'll take another look at it today. The rest of the journal-cms instances aren't going anywhere and I now have `continuumtest` to play with.

Reply:
top answer to this question looks promising: 
https://unix.stackexchange.com/questions/288236/have-systemd-spawn-n-processes

I'll give it a shot today.

Reply:
ok, have another look at what I've done. I'm going to do a few more tests after lunch to ensure basic cases are working as expected - it's difficult to know for sure with these processes on journal-cms - but it's seems like an elegant solution so far. 

If it all works out I'll do a write-up to explain the history and current state of all this and how it should happen going forwards.

Reply:
(sigh) it should work in theory but it doesn't in practice. I have a self-contained example happening in builder-base-formula that just prints a message to the log every 5 seconds. Starting the `.target` starts the services, but stopping the `.target` doesn't. It feels like the right approach so I suspect I haven't created the link dependency between target correctly

Reply:
ha! I have it working. It's always the way. Some combination of typos, bad state and misleading internet noise has one going down the wrong rabbit hole...

```
May 14 07:53:48 dev--basebox.elifesciences.org systemd[1]: Started "tester".
May 14 07:53:48 dev--basebox.elifesciences.org systemd[1]: Started "tester".
May 14 07:53:48 dev--basebox.elifesciences.org systemd[1]: Started "tester".
May 14 07:53:48 dev--basebox.elifesciences.org systemd[1]: Reached target tester-controller.target.
May 14 07:53:48 dev--basebox.elifesciences.org /tester.py[25353]: aid: [], pid: 25353, iteration: 1
May 14 07:53:48 dev--basebox.elifesciences.org /tester.py[25351]: aid: [], pid: 25351, iteration: 1
May 14 07:53:48 dev--basebox.elifesciences.org /tester.py[25352]: aid: [], pid: 25352, iteration: 1
May 14 07:53:53 dev--basebox.elifesciences.org /tester.py[25353]: aid: [], pid: 25353, iteration: 2
May 14 07:53:53 dev--basebox.elifesciences.org /tester.py[25351]: aid: [], pid: 25351, iteration: 2
May 14 07:53:53 dev--basebox.elifesciences.org /tester.py[25352]: aid: [], pid: 25352, iteration: 2
May 14 07:53:58 dev--basebox.elifesciences.org /tester.py[25353]: aid: [], pid: 25353, iteration: 3
May 14 07:53:58 dev--basebox.elifesciences.org /tester.py[25351]: aid: [], pid: 25351, iteration: 3
May 14 07:53:58 dev--basebox.elifesciences.org /tester.py[25352]: aid: [], pid: 25352, iteration: 3
May 14 07:54:03 dev--basebox.elifesciences.org systemd[1]: Stopped target tester-controller.target.
May 14 07:54:03 dev--basebox.elifesciences.org systemd[1]: Stopping "tester"...
May 14 07:54:03 dev--basebox.elifesciences.org systemd[1]: Stopping "tester"...
May 14 07:54:03 dev--basebox.elifesciences.org systemd[1]: Stopping "tester"...
May 14 07:54:03 dev--basebox.elifesciences.org systemd[1]: Stopped "tester".
May 14 07:54:03 dev--basebox.elifesciences.org systemd[1]: Stopped "tester".
May 14 07:54:03 dev--basebox.elifesciences.org systemd[1]: Stopped "tester".
```

polishing it a bit before I sign off